### PR TITLE
Allow returning `self` from getters

### LINKED ISSUE #579: Returning `self` from getters
The following contract compiles, but produces a TS error:

```tact
import "@stdlib/deploy";

contract Test with Deployable {
    id: Int as uint32 = 0;
    anotherData: Cell;

    init() {
        self.anotherData = beginCell().endCell();
    }

    get fun contractData(): Test {
        return self;
    }
}
```

Test:

```ts
import { Blockchain, SandboxContract, TreasuryContract } from '@ton/sandbox';
import { toNano } from '@ton/core';
import { Test } from '../wrappers/Test';
import '@ton/test-utils';

describe('Test', () => {
    let blockchain: Blockchain;
    let deployer: SandboxContract<TreasuryContract>;
    let test: SandboxContract<Test>;

    beforeEach(async () => {
        blockchain = await Blockchain.create();

        test = blockchain.openContract(await Test.fromInit());

        deployer = await blockchain.treasury('deployer');

        const deployResult = await test.send(
            deployer.getSender(),
            {
                value: toNano('0.05'),
            },
            {
                $$type: 'Deploy',
                queryId: 0n,
            }
        );

        expect(deployResult.transactions).toHaveTransaction({
            from: deployer.address,
            to: test.address,
            deploy: true,
            success: true,
        });
    });

    it('should get data', async () => {
        const data = await test.getContractData();
        console.log(data);
    });
});
```

### PR DESCRIPTION
**Pull request by Gusarich**
Closes #579 

- [x] I have updated CHANGELOG.md
- [ ] ~~I have documented my contribution in Tact Docs: https://github.com/tact-lang/tact-docs/pull/PR-NUMBER~~
- [x] I have added tests to demonstrate the contribution is correctly implemented: this usually includes both positive and negative tests, showing the happy path(s) and featuring intentionally broken cases
- [x] I have run all the tests locally and no test failure was reported
- [x] I have run the linter, formatter and spellchecker
- [x] I did not do unrelated and/or undiscussed refactorings

### CODE REVIEW

File: src/bindings/typescript/serializers.ts [hunk 1]
```diff
@@ -558,9 +558,9 @@ const struct: Serializer<{ name: string; optional: boolean }> = {
     },
     tsType(v) {
         if (v.optional) {
-            return v.name + " | null";
+            return v.name + "_struct | null";
```
Comment (anton-trunov) [inline src/bindings/typescript/serializers.ts]: why this change?

Comment (Gusarich) [reply src/bindings/typescript/serializers.ts]: Because I've added `_struct` suffix to struct/contract types in TS, so that they don't overlap with contract names

Comment (anton-trunov) [reply src/bindings/typescript/serializers.ts]: Hmmm, this breaks some existing tests (like the ones in `wallet.spec.ts`). I would add a prefix/suffix to a less frequent use case of the contract "struct" (like in the case of a getter returning a contract) to break as little existing test code as possible.

Comment (Gusarich) [reply src/bindings/typescript/serializers.ts]: Done!

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Changed
 
+- Returning `self` from getters is now allowed: PR [#666](https://github.com/tact-lang/tact/pull/666)
```
Comment (anton-trunov) [inline CHANGELOG.md]: Let's move this to `Fixed`

File: src/types/resolveDescriptors.ts [hunk 3]
```diff
@@ -1818,7 +1818,7 @@ export function getType(
     ident: AstId | string,
 ): TypeDescription {
     const name = typeof ident === "string" ? ident : idText(ident);
-    const r = store.get(ctx, name);
+    const r = store.get(ctx, name.endsWith("$Data") ? name.slice(0, -5) : name);
```
Comment (anton-trunov) [inline src/types/resolveDescriptors.ts]: Looks like this check should be moved to the place(s) related to ABI?

**Review by anton-trunov — CHANGES_REQUESTED** [review]

File: src/types/resolveDescriptors.ts (hunk 3 above)
Comment (Gusarich) [reply src/types/resolveDescriptors.ts]: done, have a look

**Review by anton-trunov — APPROVED** [review]
LGTM!
